client/landing/stepper/hooks/use-is-site-big-sky-eligible.ts

```diff
@@ -1,30 +1,62 @@
 import config from '@automattic/calypso-config';
-import { isPremiumPlan } from '@automattic/calypso-products';
+import { isBusinessPlan, isPremiumPlan } from '@automattic/calypso-products';
 import { useBreakpoint } from '@automattic/viewport-react';
 import { useSelect } from '@wordpress/data';
+import { useState, useEffect } from 'react';
+import { useExperiment } from 'calypso/lib/explat';
 import { useIsSiteOwner } from '../hooks/use-is-site-owner';
 import { ONBOARD_STORE } from '../stores';
 import { useSite } from './use-site';
 import type { OnboardSelect } from '@automattic/data-stores';
 
-export function useIsBigSkyEligible(): boolean | null {
+const bypassBigSkyExperiment = () => config.isEnabled( 'big-sky' );
+
+export function useIsBigSkyEligible() {
 	const { isOwner } = useIsSiteOwner();
 	const site = useSite();
-	const productSlug = site?.plan?.product_slug || '';
+	const product_slug = site?.plan?.product_slug || '';
 	const isNarrowView = useBreakpoint( '<800px' );
 	const goals = useSelect(
 		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getGoals(),
 		[ site ]
 	);
-	const validGoals = [ 'other', 'promote' ];
 
-	const configEnabled = config.isEnabled( 'calypso/ai-assembler' );
-	if ( ! configEnabled ) {
-		return false;
-	}
+	const featureFlagEnabled = config.isEnabled( 'calypso/big-sky' );
+	const [ , experimentAssignment ] = useExperiment(
+		'calypso_signup_onboarding_bigsky_soft_launch',
+		{
+			isEligible: featureFlagEnabled && ! bypassBigSkyExperiment(),
+		}
+	);
+
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ isEligible, setIsEligible ] = useState( false );
+
```

Comment:
You don't really need to remember the loading state, the only source of asynchronously is `useExperiment`, I'd do 

```js
const [ isLoading , experimentAssignment ] = useExperiment(
		'calypso_signup_onboarding_bigsky_soft_launch',
		{
			isEligible: featureFlagEnabled && ! bypassBigSkyExperiment(),
		}
	);

if ( isLoading ) {
	return { isLoading, isElligible: false } 
}

``` 

And at the end of the function return

```js
const eligibilityResult =
			( isOwner && isEligiblePlan && hasValidGoal && ! isNarrowView ) || false;
return { isLoading: false, isElligible: eligibilityResult } 
```

I promise I'm not being nitpicky, creating needless state is like caching stuff, it's the number of source of bugs. Happy to push myself if this is confusing.

Reply:
This makes a lot more sense, thanks.